.gitlab/pipeline.yml

```
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2023 ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file BOOST_LICENSE_1_0.rst or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+include:
+  - remote: 'https://gitlab.com/cscs-ci/recipes/-/raw/master/templates/v2/.ci-ext.yml'
+
+stages:
+  - build
+  - test
+
+###################################################################################################
+
+clang14_debug_build:
+  stage: build
+  extends: .container-builder
+  variables:
+    BASE_IMAGE: docker.io/pikaorg/pika-ci-base:17
+    DOCKERFILE: .gitlab/docker/Dockerfile.build
+    DOCKER_BUILD_ARGS: '["BASE_IMAGE"]'
+    PERSIST_IMAGE_NAME: $CSCS_REGISTRY_PATH/pika-debug-cpu-clang14:$CI_COMMIT_SHORT_SHA
+
+###################################################################################################
+
+.test_common:
+  variables:
+    SLURM_PARTITION: normal
+
+.test_common_mc:
+  stage: test
+  extends: .test_common
+  extends: .container-runner-daint-mc
+  variables:
+    SLURM_CONSTRAINT: mc
+    SLURM_PARTITION: normal
+    PROCESSOR_COUNT: 36
+
+.test_common_gpu:
+  extends: .test_common
+  extends: .container-runner-daint-gpu
+  variables:
+    SLURM_CONSTRAINT: gpu
+    SLURM_PARTITION: normal
+    PROCESSOR_COUNT: 12
+
+clang14_debug_test:
+  extends: .test_common_mc
+  needs:
+    - clang14_debug_build
+  image: $CSCS_REGISTRY_PATH/pika-debug-cpu-clang14:$CI_COMMIT_SHORT_SHA
+  script:
+    - cd /pika/build
```

I don't know if environment variables are passed into the container at runtime, but if they do I guess this could be
```suggestion
    - cd $BUILD_DIR
```
Not a blocker.